pkg/server/tls/tls.go

```diff
@@ -0,0 +1,91 @@
+package tls
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"math/big"
+	"net"
+	"time"
+)
+
+// TLSConfig is adapted from http.Server.ServeTLS
+func TLSConfig(hosts []string) (*tls.Config, error) {
+	certPEMBlock, keyPEMBlock, err := generateKeyPair(hosts)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to generate TLS keys %s", err)
```

Comment:
This should be `%w` and not `%s` https://pkg.go.dev/fmt#Errorf

Same issue below.